sos/plugins/ovn_central.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2018 Mark Michelson <mmichels@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import json
+import os
+
+
+class OVNCentral(Plugin):
+    """ OVN Northd
+    """
+    plugin_name = "ovn_central"
+    profiles = ('network', 'virt')
+
+    def setup(self):
+
+        all_logs = self.get_option("all_logs")
+        limit = self.get_option("log_size")
+
+        if not all_logs:
+            self.add_copy_spec("/var/log/openvswitch/*.log",
+                               sizelimit=limit)
+        else:
+            self.add_copy_spec("/var/log/openvswitch/",
+                               sizelimit=limit)
+
+        self.add_copy_spec([
+            "/var/run/openvswitch/ovnnb_db.pid",
+            "/var/run/openvswitch/ovnsb_db.pid",
+            "/var/run/openvswitch/ovn-northd.pid",
+        ])
+
+        cmds = []
+
+        with open('/usr/share/openvswitch/ovn-nb.ovsschema', 'r') as f:
+            db = json.load(f)
+            for table in db['tables'].iterkeys():
+                cmds.append('ovn-nbctl list %s' % table)
+
+        with open('/usr/share/openvswitch/ovn-sb.ovsschema', 'r') as f:
+            db = json.load(f)
+            for table in db['tables'].iterkeys():
+                # Don't include Logical_Flow because we'll use the more
+                # user-friendly `ovn-sbctl lflow-list` command to get that
+                # information
+                if table != 'Logical_Flow':
+                    cmds.append('ovn-sbctl list %s' % table)
+
+        # Some user-friendly output of DB output
+        cmds.append('ovn-sbctl lflow-list')
+        cmds.append('ovn-nbctl get-ssl')
+        cmds.append('ovn-nbctl get-connection')
+        cmds.append('ovn-sbctl get-ssl')
+        cmds.append('ovn-sbctl get-connection')
```

Comment:
Any reason for making an `append()` call for these static items, rather than simply initialising the list to contain them in the first place? That's more in keeping with existing style in sos, as well as avoiding some (admittedly trivial) runtime cost. (note that the order of the commands in the list passed to `add_cmd_output()` has no significance).

Reply:
Thanks, I'll make this change.